### ISSUE TITLE
Trouble launching

### ISSUE THREAD

**Issue by quickfold** [euu2021/Freeplane_UtilityPanels#2]
I'm finally trying this script out! When I launch from Tools/Scripts/Utility Panels, I get 3 blank panels in the upper left of the map window. I'm using FP 1.2.8. What should I try to troubleshoot?

![image](https://github.com/user-attachments/assets/f106fda1-3112-48ad-b944-380aee85782b)

**Comment by euu2021** [comment]
Please, try selecting some nodes. I.e., change the selection. It should make some changes in the first panel.

**Comment by quickfold** [comment]
When I select nodes, I see no changes. By the way, I am using Freeplane Portable.

**Comment by euu2021** [comment]
Please, try activating the script from `Tools->Edit script…`

- Create a new  script:
![image](https://github.com/user-attachments/assets/824153ee-183a-443e-b673-e07389ba515a)

- Paste the UtilityPanels code there:

![image](https://github.com/user-attachments/assets/9006cd4f-ccbb-4624-8ec2-e566b78a2dd2)

- Run
![image](https://github.com/user-attachments/assets/dce327b7-b7fa-4b0a-a0d5-6d0f383af1ca)

- Select some nodes, and they should appear on the first panel:

![image](https://github.com/user-attachments/assets/ddc995f5-2700-4c36-8511-fbdeeccae8f2)

**Comment by quickfold** [comment]
Unfortunately, I get the same result. Can you confirm that your script works on Freeplane portable? Here's what I got even after selecting different nodes:

![image](https://github.com/user-attachments/assets/92b337b6-3272-4673-b434-bfa2ab2533ba)

**Comment by euu2021** [comment]
> Can you confirm that your script works on Freeplane portable? 

Yes, I use Freeplane portable. I tried several Java versions, but I don't see that bug.

Here is what [another user said](https://github.com/euu2021/Freeplane_UtilityPanels/issues/4#issuecomment-2550339221) about that bug:

> The script worked for me. Once, when I opened a few more maps, exactly the same thing happened: the panels could not display anything. But later it was fixed again and now I have no particular problems. I don't know how the problem was fixed. 
You can see the version I installed in https://github.com/euu2021/Freeplane_UtilityPanels/issues/6 .

**Comment by quickfold** [comment]
Thanks for the reply. I really want to solve this. I'm running FP Portable 1.12.8 and Win 11 24H2. Running the utility panels script not only doesn't work but actually causes my Freeplane to stop responding (both older versions and 1.15) and I have to kill the process. What can I tell you to help me troubleshoot?

- Do you use Windows?
- Can you tell me how to find my Java version so I can give you that info, or how to install a Java version that you know works? - FYI, I've tried with different maps based on different templates.
- I looked at the other thread you linked to and the user has a script classpath set. I don't. Do I need to?
- Is there some environmental variable in Windows I need to set or something?

If you're willing to set up a time to troubleshoot live with a shared screen, let me know.

**Comment by euu2021** [comment]
I tried with a fresh profile and now I see the bug. It's related to some setting, but I don't know which one, yet.

If you try my auto.properties, I think the script will work. Go into `Tools->Open user directory`, backup your auto.properties, and replace with this one:

[auto.zip](https://github.com/user-attachments/files/18275297/auto.zip)

**Comment by quickfold** [comment]
I replaced my auto.properties with yours, launched FP (it looked very different), copied the 1.18 version of the script, put it into 'edit script' ran it, and the output windows said 'Result: null' and I got the same type of freezing/non-response as in previous times. Now that you are creating a tags panel, I want it to work even more.

**Comment by euu2021** [comment]
> I replaced my auto.properties with yours, launched FP (it looked very different), copied the 1.18 version of the script, put it into 'edit script' ran it, and the output windows said 'Result: null' and I got the same type of freezing/non-response as in previous times. Now that you are creating a tags panel, I want it to work even more.

Ok, one of the problems is that FP is finding an error when trying to create, for the first time, the configs file for the script. So, the steps to make the script work, while I don't find a solution:

- use freeplane-1.12.8-pre03 (or older);
- in your user directory folder, use my auto.properties file above;
- in your user directory folder, manually create the UtilityPanels config file. So, create a text file with this content:

```
{
    "pinnedItems": [
        
    ],
    "recentSearches": [
        ""
    ],
    "recentNodes": [
        
    ]
}

```

And then rename the file to `utilityPanelsConfig.json`. Or simply use this file:
[utilityPanelsConfig.zip](https://github.com/user-attachments/files/18285354/utilityPanelsConfig.zip)


- Save the script in your user directory folder, inside the folder `scripts`

- Start FP

- Start the script at the Euu menu that shows at the toolbar

![image](https://github.com/user-attachments/assets/65e8d6f3-ea1d-4a20-a3e1-a0ea05e103f8)

**Comment by euu2021** [comment]
Actually, the first problem (empty panels) is because of the map used. So, it's irrelevant to change the auto.properties. If you try this map and follow the steps above (skipping the auto.properties step), it should work:
[EXAMPLE template.zip](https://github.com/user-attachments/files/18285583/EXAMPLE.template.zip)

I will try to understand.

**Comment by euu2021** [comment]
Problem 1 is fixed in 1.9.

**Comment by euu2021** [comment]
Problem 1 is fixed in 1.20.

Now, if you just need to manually create utilityPanelsConfig.json, and you are ready to go.

**Comment by euu2021** [comment]
Both bugs fixed in 1.21.

You will be able to use the script straight away with freeplane-1.12.8-pre03 or older. freeplane-1.12.9 is still not compatible.

**Comment by quickfold** [comment]
Ok, it now basically works! However, I am getting a lot of logged errors. Also, the script prevents my FP from highlighting selected nodes--whatever node was highlighted/selected when I run the script remains so, even though I can see I am navigating around the map based on the 'recently selected nodes' panel, so I can't use the scipt in its current state.

Do you want to see my log file, or no need?

Also, I have many suggestions for improvements (such as labeling the panels), do you want me to create a separate issue for each idea or do it some other way?

**Comment by euu2021** [comment]
> Do you want to see my log file, or no need?

Yes, please

> Also, I have many suggestions for improvements (such as labeling the panels), do you want me to create a separate issue for each idea or do it some other way?

I think you can do an overview about your needs in a discussion thread, and then open an issue for each idea.

**Comment by quickfold** [comment]
See attached
[New Mindmap.zip](https://github.com/user-attachments/files/18295853/New.Mindmap.zip)

**Comment by euu2021** [comment]
Your auto.properties have scripts permissions disabled. Try enabling it at:

![image](https://github.com/user-attachments/assets/ec23c0a0-ea1e-4e2a-bd1e-25b0101a5915)

**Comment by quickfold** [comment]
OK, it seems to be working, thanks! Now to suggest improvements under "issues"...

**Comment by fgionghi** [comment]
I have kind of the same problem: white empty utility panel and my map also freeze and I have to kill freeplane. I am on ubuntu 24 with stable freeplane 1.28. This is the map I am using:

[test.zip](https://github.com/user-attachments/files/18440184/test.zip)

**Comment by euu2021** [comment]
Looking at your screenshot, I think it's working fine. The readme description is outdated. There was some changes since I wrote it.

The only bug that I see is that the inspectors should be more to the left, starting at the end of the left panel. That gap shouldn't be there. 

>  The problem seems with colors, right?

Do you mean, the purple and red borders around TODO item? It's correct.
-red: means that this item is selected on the map
-purple: means that there is an inspector opened corresponding to that item. In your screenshot, the first inspector refers to the TODO item, so it gets purple borders when showing on other panels.

> I am on ubuntu 24 with stable freeplane 1.28. 

Theoretically, the latest UtilityPanels versions need freeplane >= 1.29. But, if it's working for you, fine. 
For freeplane 1.28, you can get older versions at:


![Image](https://github.com/user-attachments/assets/8626b2b0-ca02-4ea9-83bd-ec059d7c575d)

**Comment by euu2021** [comment]
> The only bug that I see is that the inspectors should be more to the left, starting at the end of the left panel. That gap shouldn't be there.

Fixed on 1.31
